### PR TITLE
Fix roll back when releasing latest binaries to s3 and change condition when triggering delete-image-binaries in CD workflow

### PR DESCRIPTION
**Pull request by khanhntd**
**Description:** <Describe what has changed.>

Instead of release s3 to latest, we only need to release version s3 and release to latest if the validation test success. Moreover,  changing conditions when triggering delete-image-binaries

**Link to tracking Issue:** <Issue number if applicable>
N/A
**Testing:** <Describe what testing was performed and which tests were added.>
N/A
**Documentation:** <Describe the documentation added.>
N/A

### CODE REVIEW

File: .github/workflows/CD.yml [hunk 1]
```diff
@@ -505,9 +505,9 @@ jobs:
           retry_wait_seconds: 20
           command: cd testing-framework/terraform/eks && terraform destroy -auto-approve
 
-  delete-images-binaries-if-fail:
+  delete-images-binaries:
     runs-on: ubuntu-latest
-    if: failure()
+    if: $ {{ failure() || cancelled() }}
```
Comment (jefchien) [inline .github/workflows/CD.yml:510]:
```suggestion
    if: ${{ failure() || cancelled() }}
```

Comment (khanhntd) [reply .github/workflows/CD.yml]: Nice catch. Fixed it in the newest commit.

File: .github/workflows/CD.yml [hunk 2]
```diff
@@ -590,13 +596,17 @@ jobs:
 
       - name: Pull image from integration test ECR, tag with input version and push to DockerHub
         uses: akhilerm/tag-push-action@v2.0.0
-        if: steps.release-latest-image.outputs.cache-hit != 'true'
+        if: steps.release-latest-image-s3.outputs.cache-hit != 'true'
         with:
           src: public.ecr.aws/${{ env.ECR_REPO }}:${{ needs.release-checking.outputs.version }}
           dst: ${{ env.IMAGE_NAMESPACE }}/${{ env.IMAGE_NAME }}:${{ needs.release-checking.outputs.version }}
 
+      - name: Release binaries to s3 with latest version
```
Comment (jefchien) [inline .github/workflows/CD.yml:604]: I would keep this separate.

Comment (khanhntd) [reply .github/workflows/CD.yml]: Reasonable. Separate the release-latest s3 and release-latest image into two different jobs.

**Review by sethAmazon — APPROVED** [review]

**Review by jefchien — APPROVED** [review]

**Pull request merged**